sklearn/ensemble/tests/test_forest.py

```diff
@@ -1330,3 +1330,61 @@ def test_forest_degenerate_feature_importances():
     gbr = RandomForestRegressor(n_estimators=10).fit(X, y)
     assert_array_equal(gbr.feature_importances_,
                        np.zeros(10, dtype=np.float64))
+
+
+@pytest.mark.parametrize('name', FOREST_CLASSIFIERS_REGRESSORS)
+@pytest.mark.parametrize(
+    'max_samples_exception',
+    [(int(1e9), ValueError),
+     (1.0, ValueError),
+     (2.0, ValueError),
+     (0.0, ValueError),
+     (np.nan, ValueError),
+     (np.inf, ValueError),
+     ('str max_samples?!', TypeError),
+     (np.ones(2), TypeError)]
+)
+def test_max_samples_exceptions(name, max_samples_exception):
+    """ Check invalid `max_samples` values
+    """
+    max_samples, exception = max_samples_exception
+    est = FOREST_CLASSIFIERS_REGRESSORS[name](max_samples=max_samples)
+    with pytest.raises(exception):
```

Comment:
```suggestion
    with pytest.raises(err_type, match=err_msg):
```

Reply:
It has to work, you are only missing the `\` before `()<>` characters.
The following patch is working:

```diff
diff --git a/sklearn/ensemble/tests/test_forest.py b/sklearn/ensemble/tests/test_forest.py
index 1998cd414..ceecab1cf 100644
--- a/sklearn/ensemble/tests/test_forest.py
+++ b/sklearn/ensemble/tests/test_forest.py
@@ -1338,27 +1338,27 @@ def test_forest_degenerate_feature_importances():
     [(int(1e9), ValueError,
       "`max_samples` must be in range 1 to 6 but got value 1000000000"),
      (1.0, ValueError,
-      "`max_samples` must be in range (0, 1) but got value 1.0"),
+      r"`max_samples` must be in range \(0, 1\) but got value 1.0"),
      (2.0, ValueError,
-      "`max_samples` must be in range (0, 1) but got value 2.0"),
+      r"`max_samples` must be in range \(0, 1\) but got value 2.0"),
      (0.0, ValueError,
-      "`max_samples` must be in range (0, 1) but got value 0.0"),
+      r"`max_samples` must be in range \(0, 1\) but got value 0.0"),
      (np.nan, ValueError,
-      "`max_samples` must be in range (0, 1) but got value nan"),
+      r"`max_samples` must be in range \(0, 1\) but got value nan"),
      (np.inf, ValueError,
-      "`max_samples` must be in range (0, 1) but got value inf"),
+      r"`max_samples` must be in range \(0, 1\) but got value inf"),
      ('str max_samples?!', TypeError,
-      "`max_samples` should be int or float, but got type '<class \'str\'>'"),
+      r"`max_samples` should be int or float, but got type "
+      r"'\<class \'str\'\>'"),
      (np.ones(2), TypeError,
-      "`max_samples` should be int or float, but got type "
-      "'<class \'numpy.ndarray\'>'")]
+      r"`max_samples` should be int or float, but got type "
+      r"'\<class \'numpy.ndarray\'\>'")]
 )
 def test_max_samples_exceptions(name, max_samples, exc_type, exc_msg):
     # Check invalid `max_samples` values
     est = FOREST_CLASSIFIERS_REGRESSORS[name](max_samples=max_samples)
-    with pytest.raises(exc_type) as exc_info:
+    with pytest.raises(exc_type, match=exc_msg):
         est.fit(X, y)
-    assert str(exc_info.value) == exc_msg, "Exception message does not match"
```

Reply:
Ah, I see what you mean. I was trying to avoid that, but the strings are long and messy anyway, so :man_shrugging:  ... done in https://github.com/scikit-learn/scikit-learn/pull/14682/commits/7869f87f10a6bfeef711d53753a94a0da7d911a7